src/add_caps.jl

```diff
@@ -113,17 +112,25 @@ end
         m = length(all_versions)
         # Bound the latest version
         version = all_versions[end]
-        add_caps!(compat,
-                  deps,
-                  UpperBound(),
-                  registry_paths,
-                  pkg_to_latest_version,
-                  pkg_to_latest_zero_version,
-                  pkg,
-                  pkg_path,
-                  version)
+        if option isa CapLatestVersion
+            add_caps!(compat,
+                    deps,
+                    UpperBound(),
+                    registry_paths,
+                    pkg_to_latest_version,
+                    pkg_to_latest_zero_version,
+                    pkg,
+                    pkg_path,
+                    version)
+        elseif !isa(option, ExcludeLatestVersion())
+            error("add_caps requires ExcludeLatestVersion or CapLatestVersion option")
+        end
         # Propagate the bounds backwards
         for j = m-1:-1:1
+            if !haskey(compat, version)
+                @warn "$compat_toml is suspicious"
+                continue
```

Comment:
Won't this then discard the upper bounds on the previous versions? (Sorry, I am not fresh enough on this any more to really be sure I can predict what will happen.)

Reply:
Does this answer the question? https://github.com/bcbi/RetroCap.jl/pull/38#discussion_r375948396

Reply:
Right, but if it discards bounds that were present, is that the correct outcome?

I think the reason I insisted on CapLatestVersion was simply to "seed" the bounds for everything at the beginning of the recursive pass. If you want to support ExcludeLatestVersion I think you'll need to come up with a better way of initializing in a case where it wasn't seeded at the beginning.

Reply:
Yeah. The more I think about this, the more that I think it doesn't make sense to support monotonic bounds if you don't have `CapLatestVersion`.

Why not just use the regular `UpperBound` instead of `MonotonicBounds`?